### PR TITLE
create @magenta/protobuff, fix @magenta/core build

### PR DESCRIPTION
**Pull request by hapticdata**
I attempted to use `@magenta/core` off npm and was not able to. The file that is referenced in the _package.json_ was _dist/magenta.js_ which is a _standalone_ browserify bundle. That bundle is suitable for use via a script tag and a global variable but not via npm.

To use off npm you will want to produce es5 commonjs modules using `tsc`. I renamed that script in the package.json to `bundle` and made `build` to just run `tsc` to output the correct es5 modules; however this threw the following error:
```
tsconfig.json:8:5 - error TS5053: Option 'allowJs' cannot be specified with option 'declaration'.

8     "declaration": true,
      ~~~~~~~~~~~~~
tsconfig.json:10:5 - error TS5053: Option 'allowJs' cannot be specified with option 'declaration'.

10     "allowJs": true,
       ~~~~~~~~~
error Command failed with exit code 2.
```

You will need to split the protobuff into its own npm package unless either [this bug gets closed](https://github.com/Microsoft/TypeScript/issues/7546) or protobuffs can generate Typescript files. I think this is actually a good thing to do anyway, tiny packages in js are preferred. This enables other projects to just require the protobuff if thats all that is necessary without the filesize bloat of unused features.

I went ahead and split the projects up into _js/protobuff_ and _js/core_ this will allow you to load in the js and its type definition while still producing es5 modules with type declarations in core by disabling the `allowJs` option. 

I have tested this and successfully built core with the protobuff. The way to do this without the package published is:

```
cd protobuff/
npm link
cd ../core
npm link @magenta/protobuff
```

the first `npm link` sets that folder to a global package, the second one creates a symlink of your global package to the scope of that project.

I also snuck in exporting `tensorflow.magenta.INoteSequence` from protobuff. I recommend enabling people to use plain objects and interfaces and to write APIs allowing this.

One last thing I didn't do that would be nice, you could create `protobuff/src/index.js` (and index.d.ts), set that to the main, and export the protobuff without the `tensorflow.magenta` namespace.

### CODE REVIEW

**Comment by hapticdata** [comment]
@nsthorat
@torinmb
@halfdanj
@adarob

**Review by cghawthorne — CHANGES_REQUESTED** [review]
Thanks for the fixes! One small change: 'protobuf' (rather than 'protobuff') is the preferred abbreviation for protocol buffers. Can you switch the directory/package names to that?

File: magenta/js/core/package.json [hunk 1]
```diff
@@ -2,9 +2,11 @@
   "name": "@magenta/core",
   "version": "0.0.7",
   "description": "",
-  "main": "dist/magenta.js",
-  "types": "dist/magenta.d.ts",
-  "dependencies": {},
+  "main": "es5/index.js",
+  "types": "es5/index.d.ts",
+  "dependencies": {
+		"@magenta/protobuff": "^0.0.01"
```
Comment (cghawthorne) [inline magenta/js/core/package.json:8]: s/protobuff/protobuf/

Comment (cghawthorne) [reply magenta/js/core/package.json:8]: I'll just go ahead and merge and change this myself.

**Review by cghawthorne — APPROVED** [review]

**Comment by cghawthorne** [comment]
Thanks!

**Pull request merged**